implemented resolution 10 to 13 of call 28-02-17

maximelefrancois86 opened the pull request:
Implementation of resolutions: 

- Actuators and Sensors in SSN are made subclass of System
- There is a property in SSN that allows Sensors or Actuators to be “attached” to a Platform, pending a decision on the name
- have a generic property in SOSA that allows attaching of Sensors/Actuators to Platforms and name it hosts/isHostedBy
- have the same property in SSN that allows attaching of Sensors/Actuators and Systems to Platforms and name it hosts/isHostedBy as defined in SOSA

see inline comments in https://github.com/w3c/sdw/pull/592/files to check this pull request.

maximelefrancois86 committed:
implemented resolution 10 to 13 of call 28-02-17

maximelefrancois86 commented on ssn/integrated/oldssn.ttl:
```diff
@@ -86,13 +86,6 @@ rdfs:seeAlso rdf:type owl:AnnotationProperty .
-                   rdfs:comment "Relation between a Platform and any Systems (e.g., Sensors) that are attached to the Platform." ;
-                   rdfs:isDefinedBy "http://www.w3.org/ns/ssn" ;
-                   rdfs:label "attached system" ;
-                   rdfs:seeAlso "http://www.w3.org/2005/Incubator/ssn/wiki/SSN_Deploy#PlatformSite" .
```
remove from oldssn to put in ssnx

maximelefrancois86 commented on ssn/integrated/oldssn.ttl:
```diff
@@ -216,23 +209,6 @@ ssn:hasOutput rdf:type owl:ObjectProperty ;
-                rdfs:isDefinedBy "http://www.w3.org/ns/ssn" ;
-                rdfs:label "has property" ;
-                rdfs:seeAlso "http://www.w3.org/2005/Incubator/ssn/wiki/SSN_Skeleton#Skeleton" .
-
```
remove from oldssn to put in ssnx

maximelefrancois86 commented on ssn/integrated/oldssn.ttl:
```diff
@@ -216,23 +209,6 @@ ssn:hasOutput rdf:type owl:ObjectProperty ;
-                 rdfs:isDefinedBy "http://www.w3.org/ns/ssn" ;
-                 rdfs:label "has subsystem" ;
-                 rdfs:seeAlso "http://www.w3.org/2005/Incubator/ssn/wiki/SSN_Model#System" .
-
```
remove from oldssn to put in ssnx

maximelefrancois86 commented on ssn/integrated/oldssn.ttl:
```diff
@@ -299,15 +275,6 @@ ssn:isProducedBy rdf:type owl:ObjectProperty ;
-                 rdfs:label "is property of" ;
-                 rdfs:seeAlso "http://www.w3.org/2005/Incubator/ssn/wiki/SSN_Skeleton#Skeleton" .
-
-
```
remove from oldssn to put in ssnx

maximelefrancois86 commented on ssn/integrated/oldssn.ttl:
```diff
@@ -397,14 +364,6 @@ ssn:ofFeature rdf:type owl:ObjectProperty ;
-               rdfs:label "on platform" ;
-               rdfs:seeAlso "http://www.w3.org/2005/Incubator/ssn/wiki/SSN_Deploy#PlatformSite" .
-
-
```
remove from oldssn to put in ssnx

maximelefrancois86 commented on ssn/integrated/oldssn.ttl:
```diff
@@ -509,16 +468,6 @@ ssn:DetectionLimit rdf:type owl:Class ;
-           rdfs:label "Device" ;
-           rdfs:seeAlso "http://www.w3.org/2005/Incubator/ssn/wiki/SSN_Device#Device" .
-
-
```
remove from oldssn to put in ssnx

maximelefrancois86 commented on ssn/integrated/oldssn.ttl:
```diff
@@ -688,24 +637,6 @@ Measurement result in VIM is the measured value plus any other relevant informat
-                     rdfs:isDefinedBy "http://www.w3.org/ns/ssn" ;
-                     rdfs:label "Observation Value" ;
-                     rdfs:seeAlso "http://www.w3.org/2005/Incubator/ssn/wiki/SSN_Base#Data" .
-
```
remove in oldssn as it's already in ssnx

maximelefrancois86 commented on ssn/integrated/oldssn.ttl:
```diff
@@ -754,24 +685,6 @@ ssn:Output rdf:type owl:Class ;
-             rdfs:isDefinedBy "http://www.w3.org/ns/ssn" ;
-             rdfs:label "Platform" ;
-             rdfs:seeAlso "http://www.w3.org/2005/Incubator/ssn/wiki/SSN_Deploy#PlatformSite" .
-
```
remove from oldssn to put in ssnx

maximelefrancois86 commented on ssn/integrated/oldssn.ttl:
```diff
@@ -869,37 +782,6 @@ ssn:Sensitivity rdf:type owl:Class ;
-           rdfs:label "Sensor" ;
-           rdfs:seeAlso "http://www.w3.org/2005/Incubator/ssn/wiki/SSN_Skeleton#Skeleton" .
-
-
```
remove from oldssn to put in ssnx

maximelefrancois86 replied:
removed from oldssn to put in ssn

maximelefrancois86 commented on ssn/integrated/oldssn.ttl:
```diff
@@ -869,37 +782,6 @@ ssn:Sensitivity rdf:type owl:Class ;
-           rdfs:isDefinedBy "http://www.w3.org/ns/ssn" ;
-           rdfs:label "Sensor" ;
-           rdfs:seeAlso "http://www.w3.org/2005/Incubator/ssn/wiki/SSN_Skeleton#Skeleton" .
-
```
remove in oldssn as it's already in ssnx

maximelefrancois86 commented on ssn/integrated/oldssn.ttl:
```diff
@@ -925,28 +807,6 @@ ssn:SensorInput rdf:type owl:Class ;
-                 rdfs:label "Sensor Output" ;
-                 rdfs:seeAlso "http://www.w3.org/2005/Incubator/ssn/wiki/SSN_Skeleton#Skeleton" .
-
-
```
remove in oldssn as it's already in ssnx

maximelefrancois86 commented on ssn/integrated/sosa.ttl:
```diff
@@ -50,25 +50,75 @@ sosa:FeatureOfInterest a rdfs:Class , owl:Class ;
+  sosa:ObservableProperty a rdfs:Class , owl:Class ;
+    rdfs:label "Observable Property"@en ;
+    skos:definition """An observable quality of a FeatureOfInterest."""@en ;
+    rdfs:isDefinedBy sosa: .
```
just added some indentation here

maximelefrancois86 commented on ssn/integrated/sosa.ttl:
```diff
@@ -50,25 +50,75 @@ sosa:FeatureOfInterest a rdfs:Class , owl:Class ;
+    skos:definition "An actuatable property, i.e., characteristic, of a FeatureOfInterest."@en ;
+    skos:example "A window actuator acts by changing the state between a frame and a window. The ability of the window to be opened and closed is its ActuatableProperty."@en 
+    rdfs:isDefinedBy sosa: .
+
```
just added some indentation here

maximelefrancois86 commented on ssn/integrated/sosa.ttl:
```diff
@@ -50,25 +50,75 @@ sosa:FeatureOfInterest a rdfs:Class , owl:Class ;
+  skos:definition "A Platform is an entity that hosts other entities, particularly Sensors, Actuators and other Platforms."@en ;
+  skos:example "A post, buoy, vehicle, ship, aircraft, satellite, cell-phone, human or animal may act as platforms for (technical or biological) sensors or actuators."@en ;
+  rdfs:isDefinedBy sosa: .
+
```
exactly the same as in SOSA

maximelefrancois86 commented on ssn/integrated/sosa.ttl:
```diff
@@ -50,25 +50,75 @@ sosa:FeatureOfInterest a rdfs:Class , owl:Class ;
+    schema:domainIncludes sosa:Sensor ;
+    schema:rangeIncludes sosa:Platform ;
+    owl:inverseOf sosa:hosts ;
+    rdfs:isDefinedBy sosa: .
```
exactly the same as in SOSA

maximelefrancois86 commented on ssn/integrated/sosa.ttl:
```diff
@@ -50,25 +50,75 @@ sosa:FeatureOfInterest a rdfs:Class , owl:Class ;
+    schema:rangeIncludes sosa:Actuator ;
+    schema:rangeIncludes sosa:Sensor ;
+    owl:inverseOf sosa:isHostedBy ;
+    rdfs:isDefinedBy sosa: .
```
exactly the same as in SOSA

maximelefrancois86 commented on ssn/integrated/sosa.ttl:
```diff
@@ -50,25 +50,75 @@ sosa:FeatureOfInterest a rdfs:Class , owl:Class ;
+  skos:definition """Device, agent (including humans), or software (simulation) involved in, or implementing, a (Sensing) Procedure. Sensors respond to a stimulus, e.g., a change in the environment, or input data composed from the results of prior observations, and generate a Result. Sensors can be mounted on Platforms."""@en ;
+  skos:example "Accelerometers, gyroscopes, barometers, magnetometers, and so forth a sensors that are typically mounted on a modern smart phone (which acts as Platform). Other examples of sensors include the human eyes."@en ;
   rdfs:isDefinedBy sosa: .
 
```
exactly the same as in SOSA

maximelefrancois86 commented on ssn/integrated/sosa.ttl:
```diff
@@ -50,25 +50,75 @@ sosa:FeatureOfInterest a rdfs:Class , owl:Class ;
+  rdfs:label "Actuator"@en ;
+  skos:definition "A device that is used by, or implements, an (Actuation) Procedure that changes the state of the world."@en ;
+  skos:example "A window actuators for automatic window control."@en ;
   rdfs:isDefinedBy sosa: .
```
exactly the same as in SOSA

maximelefrancois86 commented on ssn/integrated/sosa.ttl:
```diff
@@ -50,25 +50,75 @@ sosa:FeatureOfInterest a rdfs:Class , owl:Class ;
+  rdfs:label "Observation"@en ;
+  skos:definition """Activity of carrying out an (Observation) Procedure to estimate or calculate a value of a Property of a FeatureOfInterest. Links to a Platform or Sensor to describe what made the Observation and how; links to an ObservableProperty to describe what the result is an estimate of, and to a FeatureOfInterest to detail what that property was associated with. The Result is the outpu ...
+  skos:example """The activity of estimating the intensity of an Earthquake using the Mercalli intensity scale is an Observation as is measuring the moment magnitude, i.e., the energy released by said earthquake."""@en ;
+  rdfs:isDefinedBy sosa: .
```
exactly the same as in SOSA

maximelefrancois86 commented on ssn/integrated/sosa.ttl:
```diff
@@ -50,25 +50,75 @@ sosa:FeatureOfInterest a rdfs:Class , owl:Class ;
+  rdfs:label "Actuation"@en ;
+  skos:definition """An Actuation carries out an (Actuation) Procedure to change the state of the world using an Actuator."""@en ;
+  skos:example """The activity of automatically closing a window if the temperature in a room drops below 20 degree Celsius. The activity is the Actuation and the device that closes the window is the Actuator. The Procedure is the rule, plan, or specification that defines the conditions that triggers the Actuation, here a drop in temperature. """@en ;
+  rdfs:isDefinedBy sosa: .
```
exactly the same as in SOSA

maximelefrancois86 commented on ssn/integrated/ssn.ttl:
```diff
@@ -46,14 +46,66 @@ ssn: a owl:Ontology ;
+    rdfs:label "is property of"@en ;
+    skos:definition """Relation between a FeatureOfInterest and a Property (a Quality observable by a sensor) of that feature."""@en ;
+    owl:inverseOf ssn:hasProperty ;
+    rdfs:isDefinedBy ssn: .
```
added very simple ssn:hasProperty and ssn:isPropertyOf, as these properties will probably not be in SOSA

maximelefrancois86 commented on ssn/integrated/ssn.ttl:
```diff
@@ -46,14 +46,66 @@ ssn: a owl:Ontology ;
+#  rdfs:subClassOf [ owl:onProperty ssn:hasOperatingRange ; owl:allValuesFrom ssn:OperatingRange ] ;
+#  rdfs:subClassOf [ owl:onProperty ssn:hasSurvivalRange ; owl:allValuesFrom ssn:SurvivalRange ] ;
+  rdfs:isDefinedBy ssn: .
+
```
in system, deleted the existential restriction as in https://www.w3.org/2015/spatial/wiki/Platform
replaced oldssn terms by new sosa/ssn terms,
commented the axioms that involve some terms that have not yet been integrated.

maximelefrancois86 commented on ssn/integrated/ssn.ttl:
```diff
@@ -46,14 +46,66 @@ ssn: a owl:Ontology ;
+    rdfs:label "has subsystem"@en ;
+    skos:definition """Relation between a system and its component parts (has-part)."""@en ;
+    rdfs:isDefinedBy ssn: .
+
```
just put hasSubSystem here, didn't modify anything from old ssn

maximelefrancois86 commented on ssn/integrated/ssn.ttl:
```diff
@@ -46,14 +46,66 @@ ssn: a owl:Ontology ;
+
+sosa:Platform 
+  rdfs:subClassOf [ owl:onProperty sosa:hosts ; owl:allValuesFrom ssn:System ] .
+#  rdfs:subClassOf [ owl:onProperty ssn:inDeployment ; owl:allValuesFrom ssn:Deployment ] ;
```
use sosa:Platform, 
replaced oldssn terms by new sosa/ssn terms,
commented the axioms that involve some terms that have not yet been integrated.

maximelefrancois86 commented on ssn/integrated/ssn.ttl:
```diff
@@ -46,14 +46,66 @@ ssn: a owl:Ontology ;
+#  rdfs:subClassOf [ owl:onProperty ssn:detects ; owl:allValuesFrom ssn:Stimulus ] ;
+#  rdfs:subClassOf [ owl:onProperty ssn:hasMeasurementCapability ; owl:allValuesFrom ssn:MeasurementCapability ] ;
+#  rdfs:subClassOf [ owl:onProperty ssn:observes ; owl:allValuesFrom ssn:Property ] .
+
```
use sosa:Sensor,
commented the axioms that involve some terms that have not yet been integrated.

maximelefrancois86 commented on ssn/integrated/ssn.ttl:
```diff
@@ -46,14 +46,66 @@ ssn: a owl:Ontology ;
+sosa:Actuator rdfs:subClassOf ssn:System .
+#  rdfs:subClassOf [ owl:onProperty ssn:implements ; owl:someValuesFrom ssn:Acting ] ;
+#  rdfs:subClassOf [ owl:onProperty ssn:hasActuationCapability ; owl:allValuesFrom ssn:ActuationCapability ] ;
+#  rdfs:subClassOf [ owl:onProperty ssn:actsOn ; owl:allValuesFrom ssn:Property ] .
```
use sosa:Actuator,
proposed some axioms to do some parallel with sosa;Sensor, 
these are commented for now, because they involve some terms that have not yet been integrated, and we need to discuss them individually.

maximelefrancois86 commented on ssn/integrated/ssnx.ttl:
```diff
@@ -76,6 +76,98 @@ oldssn:Property a owl:Class ;
+    owl:deprecated true ;
+    owl:equivalentProperty ssn:hasProperty ; ### or sosa:hasProperty? see https://www.w3.org/2015/spatial/wiki/Link_between_FeatureOfInterest_and_xxxProperty
+    rdfs:isDefinedBy <http://purl.oclc.org/NET/ssnx/ssn> .
+
```
oldssn:hasProperty is equivalent to ssn:hasProperty

maximelefrancois86 commented on ssn/integrated/ssnx.ttl:
```diff
@@ -76,6 +76,98 @@ oldssn:Property a owl:Class ;
+    owl:deprecated true ;
+    owl:equivalentProperty ssn:isPropertyOf ; ### or sosa:isPropertyOf? see https://www.w3.org/2015/spatial/wiki/Link_between_FeatureOfInterest_and_xxxProperty
+    rdfs:isDefinedBy <http://purl.oclc.org/NET/ssnx/ssn> .
+
```
oldssn:isPropertyOf is equivalent to ssn:isPropertyOf

maximelefrancois86 commented on ssn/integrated/ssnx.ttl:
```diff
@@ -76,6 +76,98 @@ oldssn:Property a owl:Class ;
+  owl:deprecated true ;
+  owl:equivalentClass ssn:System ;
+  rdfs:isDefinedBy <http://purl.oclc.org/NET/ssnx/ssn> .
+
```
oldssn:System is equivalent to ssn:System

maximelefrancois86 commented on ssn/integrated/ssnx.ttl:
```diff
@@ -76,6 +76,98 @@ oldssn:Property a owl:Class ;
+  owl:deprecated true ;
+  owl:equivalentClass sosa:Sensor ;
+  rdfs:isDefinedBy <http://purl.oclc.org/NET/ssnx/ssn> .
+
```
oldssn:Sensor is equivalent to sosa:Sensor

maximelefrancois86 commented on ssn/integrated/ssnx.ttl:
```diff
@@ -76,6 +76,98 @@ oldssn:Property a owl:Class ;
+    rdfs:subClassOf ssn:System ;
+    rdfs:isDefinedBy "http://www.w3.org/ns/ssn" ;
+    rdfs:seeAlso "http://www.w3.org/2005/Incubator/ssn/wiki/SSN_Device#Device" .
+
```
oldssn:Device is equivalent to ssn:Device

maximelefrancois86 commented on ssn/integrated/ssnx.ttl:
```diff
@@ -76,6 +76,98 @@ oldssn:Property a owl:Class ;
+    rdfs:seeAlso "http://www.w3.org/2005/Incubator/ssn/wiki/SSN_Deploy#PlatformSite" .
+    owl:deprecated true ;
+    owl:equivalentProperty sosa:hosts ;
+    rdfs:isDefinedBy <http://purl.oclc.org/NET/ssnx/ssn> .
```
oldssn:attachedSystem is equivalent to sosa:hosts

maximelefrancois86 commented on ssn/integrated/ssnx.ttl:
```diff
@@ -76,6 +76,98 @@ oldssn:Property a owl:Class ;
+    rdfs:seeAlso "http://www.w3.org/2005/Incubator/ssn/wiki/SSN_Deploy#PlatformSite" ;
+    owl:deprecated true ;
+    owl:equivalentProperty sosa:isHostedBy ;
+    rdfs:isDefinedBy <http://purl.oclc.org/NET/ssnx/ssn> .
```
oldssn:onPlatform is equivalent to sosa:isHostedBy

maximelefrancois86 commented on ssn/integrated/ssnx.ttl:
```diff
@@ -76,6 +76,98 @@ oldssn:Property a owl:Class ;
+  rdfs:seeAlso "http://www.w3.org/2005/Incubator/ssn/wiki/SSN_Deploy#PlatformSite" ;
+  owl:deprecated true ;
+  owl:equivalentClass sosa:Platform ;
+  rdfs:isDefinedBy <http://purl.oclc.org/NET/ssnx/ssn> .
```
oldssn:Platform is equivalent to sosa:Platform

maximelefrancois86 committed:
corrected minor errors

arminhaller commented:
Thanks for kicking this off. Please do liaise with Kerry on this one, as she was assigned the action to implement this according to Action: https://www.w3.org/2015/spatial/track/actions/275

I do see several issues in your commits:
You introduce a sosa:SensorOutput. There was no decision made on this.
The propertyRestrictions on Platform, e.g. attachedSystem should probably not be removed, but changed to hosts/hostedBy as this property was renamed to.
The property restriction on Property is a seperate issue and I would not include it in this commit.
The property restrictions on System should probably also remain, but use different property names.

sosa SensorOutput, sosa SensorInput and Sosa Sensordatasheet are introduced without any decision on those.

Why is actuatableproperty changed to actuator in SOSA? We decided on ActuatableProperty already to be included in SOSA and we will have a SOSA actuator.

maximelefrancois86 committed:
corrected sosa:SensorXxx -> ssn:SensorXxx

maximelefrancois86 commented:
Hi Armin, all,

By the way, I just got noticed that everyone receives email notifications from GitHub everytime I add a comment to my own pull requests. That's something like more that 30 emails in a row yesterday... I'm sorry about this. 


Some answers below:

1. sosa:SensorOutput, sosa:SensorInput, sosa:SensorDatasheet --> my mistake, side effect of a search/replace ssn:Sensor by sosa:Sensor. Corrected.


2. The property restrictions on Platform are:

```
sosa:Platform 
  rdfs:subClassOf [ owl:onProperty sosa:hosts ; owl:allValuesFrom ssn:System ] .
#  rdfs:subClassOf [ owl:onProperty ssn:inDeployment ; owl:allValuesFrom ssn:Deployment ] ;
```

- the first is actually changed to the new term sosa:hosts;
- the second awaits decisions about deployment.


3. the property restrictions on Property is:

```
ssn:Property rdf:type owl:Class ;
#  rdfs:subClassOf [ owl:onProperty sosa:isPropertyOf ; owl:someValuesFrom sosa:FeatureOfInterest ] ; # TODO: need resolution, see https://www.w3.org/2015/spatial/wiki/Link_between_FeatureOfInterest_and_xxxProperty
```

It is commented out, waiting for resolution about ssn:hasProperty.


4. the property restrictions on System are taken from https://www.w3.org/2015/spatial/wiki/Platform  and changed to use the new term sosa:isHostedBy :

```
ssn:System rdf:type owl:Class ;
  rdfs:subClassOf [ owl:onProperty sosa:isHostedBy ; owl:allValuesFrom sosa:Platform ] ;
  rdfs:subClassOf [ owl:onProperty ssn:hasSubSystem ; owl:allValuesFrom ssn:System ] ;
#  rdfs:subClassOf [ owl:onProperty ssn:hasDeployment ; owl:allValuesFrom ssn:Deployment ] ;
#  rdfs:subClassOf [ owl:onProperty ssn:hasOperatingRange ; owl:allValuesFrom ssn:OperatingRange ] ;
#  rdfs:subClassOf [ owl:onProperty ssn:hasSurvivalRange ; owl:allValuesFrom ssn:SurvivalRange ] ;
```

- the first one has been changed to use the new term sosa:isHostedBy
- the second one can be commented out if you juge it's not mature yet
- the other ones are commented out waiting for decisions about deployment, and operating and survival properties.



5. actuatable property and actuator: I think the diff brings some confusion because there is absolutely no change in sosa.ttl about those. 


Shall I change anything else ?

Best, 
Maxime

arminhaller commented:
Yes, we do get an email for each comment. Maybe we can change the setting
That looks all good to me now.
I would actually recommend to not comment out the things we have not decided on. They are essentially unchanged from the old SSN for the time being, unless otherwise decided.
Anyway, Kerry, please have a look and do merge the pull request, otherwise I will do it before our meeting tomorrow.

KerryLea commented:
Maxime, Are your restrictions sound turtle? They don't look right to me --- don't you need to use 'rdf:type owl:Restriction" everywhere as they were in ssn.ttl?

KerryLea commented:
sorry -- di dnot mean to close -- only to comment so I have reopened the pr

maximelefrancois86 committed:
explicited 'a owl:Restriction' for every instance of OWL Restriction

maximelefrancois86 committed:
Merge branch 'gh-pages' into patch-call-2802-resolution-10-to-13

KerryLea commented:
I really appreciate you doing this -- I could not have got to it so fast. I have really only studied ssn_separated/ssn.ttl because the other  4 files are essentially your own working copies, as I understand it. 

With ref to "the second one can be commented out if you juge it's not mature yet" -- that is fine, it is mature and should be as you have it.

I agree with Armin: "I would actually recommend to not comment out the things we have not decided on. They are essentially unchanged from the old SSN for the time being, unless otherwise decided.", and also some of the commented out bits are things that were in ssn that you might be intending to remove, and others are bits that were not in ssn that you might be intending to add --- and  we can't track the difference! anyway, I can wear that, but it would be great if you could avoid  doing it  next time, please!

-- For the ssnx -- I haven't really looked but I had seen what you were doing earlier so I expect this is good

--The dul alignment still needs fixing for system (it can no longer be a physical object as it was before because now sytems can be sensors and sensors are now dul:Objects (check). This is not an objection -- just a note that it must happen before the action-275 is closed.

-- the changes to ssn here need to go in the spec as changes (I can do that). Similarly -- no object just a note before the action-275 is closed.

Am happy to merge subject to at least some reassurance about the syntax for restrictions (comment above).  Does it parse as OWL-DL?

dr-shorthair commented:
> don't you need to use 'rdf:type owl:Restriction" everywhere as they were in ssn.ttl?

It is conventional to include the rdf:type in the blank node, but OWA and RDFS entailments mean it is not strictly mandatory.

maximelefrancois86 commented:
Dear Kerry, In https://github.com/w3c/sdw/pull/592 ,  you ask:

> Maxime, Are your restrictions sound turtle? They don't look right to me
--- don't you need to
> use 'rdf:type owl:Restriction" everywhere as they were in ssn.ttl?

Well, that was a very good question, and the short answer is:
- the way I lighten the restrictions are ok for most of the OWL-based tools
I have used so far,
- but you are right, I double checked the RECs for OWL1-to-RDF [1] and
OWL2-to-RDF [2] mappings, and both seem to require that there is a "rdf:type
owl:Restriction" triple.


As far as I can remember, I always lightened the RDF document that
introduce OWL ontologies instances of owl:Restriction. I think my initial
decision to do so was motivated by:
 - the domain of owl:onProperty in https://www.w3.org/2002/07/owl# is
owl:Restriction
 - OWLAPI (and therefore Protégé) and most of the OWL editors I ever used
do recognize these lightened restrictions,
 - they are more readable to me.

So thank you for your question, it made me take some time to double check
my old habits.
The normative parts of the OWL and OWL2 RECs that describe the mapping
between the OWL abstract syntax and the RDF data model are [1] and [2],
respectively. Both require that there is a "rdf:type owl:Restriction"
triple.


By the way, Table 5 in [2] shows how OWL 2 officially deals with SOSA
classes that are declared as both rdfs:Class AND owl:Class.

I added these triples in the following commit, and appended this to the
pull request:

https://github.com/w3c/sdw/pull/592/commits/8247fcaa9922815df2477a9665116480c3603e2d


Kind regards,
Maxime

[1] - https://www.w3.org/TR/owl-semantics/mapping.html
[2] -
https://www.w3.org/TR/owl2-mapping-to-rdf/#Mapping_from_RDF_Graphs_to_the_Structural_Specification
(see
Table 13. Parsing of Class Expressions)

KerryLea commented:
all good -- much happpier now. Pls don't let me forget the dul alignment and the change log in the spec.

maximelefrancois86 commented:
Maxime, others,


On 07/03/2017 13:08, Maxime Lefrançois wrote:
>
> [skip]
 >
>
> As far as I can remember, I always lightened the RDF document that
> introduce OWL ontologies instances of owl:Restriction. I think my
> initial decision to do so was motivated by:
>   - the domain of owl:onProperty in https://www.w3.org/2002/07/owl# is
> owl:Restriction

The file that you are linking to says at the top in a literal, among 
other things:

"the descriptions provided in this ontology *do not provide a complete 
and correct formal description* of either the syntax or the semantics of 
the introduced terms (please see the OWL 2 recommendations for the 
complete and normative specifications). Furthermore, the information 
provided by this ontology may be misleading if not used with care."

(my emphasis)

This file is of no relevance to anything when discussing things related 
to OWL. It is meant to help RDF/Linked Data processors that are dealing 
with OWL terms.


It is good emphasising this every so often because there are every once 
in a while people referring to this file as a justification for dealing 
with OWL the way they do.


HTH,
--AZ

maximelefrancois86 commented:
Antoine -- all resolved happily, and thankyou kindly for your comments to keep us on the straight and narrow!-Kerry

-----Original Message-----
From: Antoine Zimmermann [mailto:antoine.zimmermann@emse.fr] 
Sent: Tuesday, 7 March 2017 11:42 PM
To: Maxime Lefrançois <maxime.lefrancois@emse.fr>; w3c/sdw <reply@reply.github.com>; w3c/sdw <sdw@noreply.github.com>; SDW WG Public List <public-sdw-wg@w3.org>
Cc: Author <author@noreply.github.com>
Subject: Re: [w3c/sdw] implemented resolution 10 to 13 of call 28-02-17 (#592)

Maxime, others,


On 07/03/2017 13:08, Maxime Lefrançois wrote:
>
> [skip]
 >
>
> As far as I can remember, I always lightened the RDF document that 
> introduce OWL ontologies instances of owl:Restriction. I think my 
> initial decision to do so was motivated by:
>   - the domain of owl:onProperty in https://www.w3.org/2002/07/owl# is 
> owl:Restriction

The file that you are linking to says at the top in a literal, among other things:

"the descriptions provided in this ontology *do not provide a complete and correct formal description* of either the syntax or the semantics of the introduced terms (please see the OWL 2 recommendations for the complete and normative specifications). Furthermore, the information provided by this ontology may be misleading if not used with care."

(my emphasis)

This file is of no relevance to anything when discussing things related to OWL. It is meant to help RDF/Linked Data processors that are dealing with OWL terms.


It is good emphasising this every so often because there are every once in a while people referring to this file as a justification for dealing with OWL the way they do.


HTH,
--AZ
